tinytga/src/header.rs

```diff
@@ -1,13 +1,63 @@
 use crate::parse_error::ParseError;
 use nom::{
-    bytes::complete::take,
-    combinator::map_res,
+    combinator::{map, map_opt, map_res},
     number::complete::{le_u16, le_u8},
     IResult,
 };
 
-/// TGA footer length in bytes
-pub const HEADER_LEN: usize = 18;
+/// Bits per pixel.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[non_exhaustive]
+pub enum Bpp {
```

Comment:
It wouldn't be very efficient to use TGA for these bit depths, because each pixel consists of an integer number of bytes. But I guess it should be possible to allow this anyway. I'll add it to the TODO for the next PR.